tests/cnf/core/network/policy/tests/multinetpolicy_bondoversriov.go

```diff
@@ -0,0 +1,720 @@
+		{
+			Name:             net1Name,
+			InterfaceRequest: "net1",
+		},
+		{
+			Name:             net2Name,
+			InterfaceRequest: "net2",
+		},
+		{
+			Name:             "bond",
+			InterfaceRequest: "bond1",
+			IPRequest:        []string{testData[podName].IPv4, testData[podName].IPv6},
+		},
+	}
+
+	tPodBuilder := pod.NewBuilder(APIClient, podName, nsName, NetConfig.CnfNetTestContainer).
+		WithNodeSelector(map[string]string{"kubernetes.io/hostname": "worker-0"}).
+		WithSecondaryNetwork(netAnnotation).
+		WithPrivilegedFlag().
+		WithLabel("app", podName)
+
+	for index, protocolPort := range testData[podName].ProtocolPort {
+		containerBuilder, err := pod.NewContainerBuilder(strings.ReplaceAll(protocolPort, ":", ""),
+			NetConfig.CnfNetTestContainer,
+			[]string{"/bin/bash", "-c", fmt.Sprintf("testcmd -listen -interface bond1 -protocol %s -port %s",
+				strings.Split(protocolPort, ":")[0], strings.Split(protocolPort, ":")[1])}).
+			WithSecurityContext(&securityContext).
+			GetContainerCfg()
+		Expect(err).ToNot(HaveOccurred(), "Failed to get container config")
+
+		if index == 0 {
+			tPodBuilder.RedefineDefaultContainer(*containerBuilder)
+		} else {
+			tPodBuilder.WithAdditionalContainer(containerBuilder)
+		}
+	}
+
+	tPod, err := tPodBuilder.CreateAndWaitUntilRunning(1 * time.Minute)
+	Expect(err).ToNot(HaveOccurred(), "Failed to create test pod")
+
+	return tPod
+}
+
+func defineNmapValidPaths(sPods, dPods []*pod.Builder, expectedResult []int64, testData podsData) {
+	By("Deriving applicable paths between given source and destination pods")
+
+	resultIndex := 0
+
+	for _, sPod := range sPods {
+		for _, dPod := range dPods {
+			// Convert expectedResult to Binary format string and do zero padding if the length of binary string is not
+			// equal to number of available paths. i.e., twice (ipv4 and ipv6) the no. of ports a pod listens to.
+			resultBinary := strconv.FormatInt(expectedResult[resultIndex], 2)
+			if len(resultBinary) < 2*len(testData[dPod.Object.Name].ProtocolPort) {
+				resultBinary = strings.Join([]string{strings.Repeat("0",
+					2*len(testData[dPod.Object.Name].ProtocolPort)-len(resultBinary)), resultBinary}, "")
+			}
+
+			resultIndex++
+
+			nmapPorts, nmapProtocols := defineNmapPortsProtocols(testData[dPod.Object.Name].ProtocolPort)
+
+			runNmapAndValidateResults(sPod, testData[sPod.Object.Name].IPv4, nmapProtocols, nmapPorts,
+				strings.Split(testData[dPod.Object.Name].IPv4, "/")[0], resultBinary[0:len(nmapPorts)])
+			runNmapAndValidateResults(sPod, testData[sPod.Object.Name].IPv6, nmapProtocols, nmapPorts,
+				strings.Split(testData[dPod.Object.Name].IPv6, "/")[0], resultBinary[len(nmapPorts):])
+		}
+	}
+}
+
+func defineNmapPortsProtocols(ports []string) ([]string, []string) {
+	var nmapPorts, nmapProtocols []string
+
+	for _, port := range ports {
+		nmapPort := strings.ToUpper(string(strings.Split(port, ":")[0][0])) + ":" + strings.Split(port, ":")[1]
```

Comment:
Please,  either simplify this logic or add a comment to the function to explain what it does. An input/output example would be very helpful

Reply:
sure, i added comment with example